frontend/src/components/editor/chrome/panels/packages-panel.tsx

```diff
@@ -157,8 +196,63 @@ export const PackagesPanel: React.FC = () => {
 
   return (
     <div className="flex-1 flex flex-col overflow-hidden">
-      <InstallPackageForm packageManager={packageManager} onSuccess={refetch} />
-      <PackagesList packages={packages} onSuccess={refetch} />
+      <InstallPackageForm
+        packageManager={packageManager}
+        onSuccess={handleRefetch}
+      />
+      {isTreeSupported && (
+        <div className="flex items-center justify-between px-2 py-1 border-b">
+          <div className="flex gap-1">
+            <button
+              type="button"
+              className={cn(
+                "px-2 py-1 text-xs rounded",
+                viewMode === "list"
+                  ? "bg-accent text-accent-foreground"
+                  : "text-muted-foreground hover:text-foreground",
+              )}
+              onClick={() => setViewMode("list")}
+            >
+              List
+            </button>
+            <button
+              type="button"
+              className={cn(
+                "px-2 py-1 text-xs rounded",
+                viewMode === "tree"
+                  ? "bg-accent text-accent-foreground"
+                  : "text-muted-foreground hover:text-foreground",
+              )}
+              onClick={() => setViewMode("tree")}
+            >
+              Tree
+            </button>
+          </div>
+          <div className="flex items-center gap-2">
+            <div
+              className="items-center border px-2 py-0.5 text-xs transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 text-foreground rounded-sm text-ellipsis block overflow-hidden max-w-fit font-medium"
+              title={treeData?.tree?.name === "<root>" ? "sandbox" : "project"}
+            >
+              {treeData?.tree?.name === "<root>" ? "sandbox" : "project"}
+            </div>
+            {treeData?.tree?.name !== "<root>" && treeData?.tree?.name && (
+              <span className="text-xs text-muted-foreground">
+                {treeData.tree.name}
+                {treeData?.tree?.version && ` v${treeData.tree.version}`}
+              </span>
```

Comment:
nit: id maybe pull out some of this into the render portion of this function.

e.g. 
```
const name = treeData?.tree?.name;
const version = treeData?.tree?.version;
```